sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -888,6 +888,22 @@ class DataFrame private[sql](
   }
 
   /**
+   * Returns a new [[DataFrame]] without duplicates.
+   * @group dfops
+   */
+  def dropDuplicates(): DataFrame = dropDuplicates(this.columns)
+
+  /**
+   * Returns a new [[DataFrame]] without duplicates under the given columns.
+   * @group dfops
+   */
+  def dropDuplicates(subset: Seq[String]): DataFrame = {
+    import org.apache.spark.sql.functions.{first => columnFirst}
+    val columnFirsts = columns.map(columnFirst)
+    groupBy(subset.head, subset.tail : _*).agg(columnFirsts.head, columnFirsts.tail : _*)
```

Comment:
we should also check if subset.size == 0 or columns.size == 0, then simply return an empty data frame (there is one in SQLContext).